### PR TITLE
ISPN-7779 State transfer does not work with protobuf encoded entities

### PR DESCRIPTION
**Pull request by anistor**
* we need to ensure __protobuf_metadata cache is started and ST complete before any cache using protobuf gets started

https://issues.jboss.org/browse/ISPN-7779

### CODE REVIEW

**Comment by tristantarrant** [comment]
I think that any Registry cache should behave like this.

**Comment by gustavonalle** [comment]
@tristantarrant +1, we are discussing that the internal caches should always starts before the others

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java [hunk 1]
```diff
@@ -117,8 +117,12 @@ private void unregisterProtobufMetadataManagerMBean(GlobalComponentRegistry gcr)
     */
    @Override
    public void cacheStarting(ComponentRegistry cr, Configuration cfg, String cacheName) {
-      InternalCacheRegistry icr = cr.getGlobalComponentRegistry().getComponent(InternalCacheRegistry.class);
+      GlobalComponentRegistry gcr = cr.getGlobalComponentRegistry();
+      InternalCacheRegistry icr = gcr.getComponent(InternalCacheRegistry.class);
       if (!icr.isInternalCache(cacheName)) {
+         ProtobufMetadataManagerImpl protobufMetadataManager = (ProtobufMetadataManagerImpl) gcr.getComponent(ProtobufMetadataManager.class);
+         protobufMetadataManager.addCacheDependency(cacheName);
+
```
Comment (gustavonalle) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java]: Shouldn't you force the start of the ProtobufMetadata cache somewhere?

Comment (anistor) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java]: that happens inside ```addCacheDependency```

Comment (gustavonalle) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java]: right, overlooked that

**Comment by anistor** [comment]
Updated. Made ProtobufMetadataManagerImpl.protobufSchemaCache volatile.

**Comment by anistor** [comment]
Should we add a new Jira to ensure internal caches are started before all other caches?

**Comment by gustavonalle** [comment]
merged, thanks

**Pull request merged**